admin/tool/policy/classes/api.php

```diff
@@ -480,29 +502,41 @@ public static function get_acceptances($versions, $select = '', $params = [], $o
 
     /**
      * Returns list of acceptances for this user.
+     * TODO: Review if it's possible to re-use some of the acceptance functions.
      *
+     * @param int $userid id of a user.
      * @param int|array $versions list of policy versions.
-     * @param int|array $userid id of a user.
      * @return array list of acceptances indexed by versionid.
      */
-    public static function get_user_acceptances($versions, $userid = null) {
+    public static function get_user_acceptances($userid, $versions = null) {
         global $DB, $USER;
-        if (!isloggedin() || isguestuser()) {
-            return;
-        }
+
         if (!$userid) {
             $userid = $USER->id;
         }
-        $usercontext = \context_user::instance($userid);
-        if ($userid != $USER->id) {
-            require_capability('tool/policy:acceptbehalf', $usercontext);
+
+        list($vsql, $vparams) = ['', []];
+        if (!empty($versions)) {
+            $versions = is_array($versions) ? array_values($versions) : [$versions];
+            list($vsql, $vparams) = $DB->get_in_or_equal($versions, SQL_PARAMS_NAMED, 'ver');
+            $vsql = ' AND a.policyversionid ' . $vsql;
         }
 
-        $acceptances = static::get_acceptances($versions);
-        if (array_key_exists($userid, $acceptances)) {
-            return $acceptances[$userid];
+        $sql = "SELECT u.id AS mainuserid, a.*, u.policyagreed
+                  FROM {user} u
+                  LEFT OUTER JOIN {tool_policy_acceptances} a ON a.userid = u.id AND a.userid = :userid $vsql";
```

Comment:
* OUTER should not be here (see the discussion in pull request #30 for background)
* `a.*` is discouraged, always name all the columns you want to get.